src/lua/ULuaScreenSing.pas

```diff
@@ -98,11 +150,41 @@ function ULuaScreenSing_SetSettings(L: Plua_State): Integer; cdecl;
       | Text: string      - text of this fragment }
 function ULuaScreenSing_GetSongLines(L: Plua_State): Integer; cdecl;
 
+function ULuaScreenSing_GetPlayerNotes(L: Plua_State): Integer; cdecl;
+
+{ THESE SHOULD BE IN ULuaScreenSong INSTEAD }
+
+{ ScreenSong.GetVisibleSongs - no arguments
+  returns number of visible songs }
+  function ULuaScreenSong_GetVisibleSongs(L: Plua_State): Integer; cdecl;
+{ ScreenSong.GetSelectedIndex - no arguments
+  returns index of currently selected song [0...visible songs -1] }
+  function ULuaScreenSong_GetSelectedIndex(L: Plua_State): Integer; cdecl;
+{ ScreenSong.GetSongNumbers - no arguments
+  returns "<selected> / <total>" }
+  function ULuaScreenSong_GetSongNumbers(L: Plua_State): Integer; cdecl;
+{ ScreenSong.GetSelectedSong - no arguments
+  returns Artist,Title,Year of currently selected song }
+  function ULuaScreenSong_GetSelectedSong(L: Plua_State): Integer; cdecl;
+
 const
-  ULuaScreenSing_Lib_f: array [0..UIni.IMaxPlayerCount-1] of lual_reg = (
+  ULuaScreenSing_Lib_f: array [0..29] of lual_reg = (
```

Comment:
where does this magic number `29` come from? Please add a code comment or (even better) move this magic number to the top of the class as a static number.

Reply:
@barbeque-squared *bump*

Reply:
29 is just the length of the array (it's either 29 or 30, I guess). If there's a way to do this without using numbers in the first place, that's by far the best solution, otherwise a code comment should suffice (I am _against_ moving this to a constant, because literally no part of the code except the array initializer cares about its size). Does Pascal have a way of instantiating these things without hardcoded numbers?

And even then, I'd argue that a hardcoded number would be more explanatory than using a _completely unrelated_ constant.

(github never gave me any notification about this)